code/modules/power/breaker_box.dm

```diff
@@ -0,0 +1,104 @@
+// AI has ability to toggle it in 5 seconds
+// Humans need 30 seconds
+// Used for advanced grid control (read: Substations)
+
+/obj/machinery/power/breakerbox
+	name = "Breaker Box"
+	icon = 'icons/obj/power.dmi'
+	icon_state = "bbox_on"
+	var/icon_state_on = "bbox_on"
+	var/icon_state_off = "bbox_off"
+	density = 1
+	anchored = 1
+	var/on = 1
+	var/busy = 0
+	var/directions = list(1,2,4,8,5,6,9,10)
+	var/update_locked = 0
+
+/obj/machinery/power/breakerbox/Initialize()
+	. = ..()
+	set_state(1)
+
+/obj/machinery/power/breakerbox/examine(mob/user)
+	. = ..()
+	to_chat(user, "Large machine with heavy duty switching circuits used for advanced grid control")
+	if(on)
+		to_chat(user, "<span class='notice'>It seems to be online.</span>")
+	else
+		to_chat(user, "<span class='warning'>It seems to be offline.</span>")
+
+/obj/machinery/power/breakerbox/attack_ai(mob/user)
+	if(update_locked)
+		to_chat(user, "<span class='warning'>System locked. Please try again later.</span>")
+		return
+
+	if(busy)
+		to_chat(user, "<span class='warning'>System is busy. Please wait until current operation is finished before changing power settings.</span>")
+		return
+
+	busy = 1
+	to_chat(user, "<span class='good'>Updating power settings..</span>")
+	if(do_after(user, 50, src))
+		set_state(!on)
+		to_chat(user, "<span class='good'>Update Completed. New setting:[on ? "on": "off"]</span>")
+		update_locked = 1
+		spawn(50) //AI isn't slow
+			update_locked = 0
+	busy = 0
+
+/obj/machinery/power/breakerbox/attack_hand(mob/user)
```

Comment:
if they're going to have functionaly identical code make them both point to the same proc instead of 2 copy pasted ones.